object_oriented_programming.ipynb

```diff
@@ -951,58 +1009,50 @@
   },
   {
    "cell_type": "markdown",
-   "id": "4abfe0ba-c893-4ffd-b40b-5931821bcc62",
-   "metadata": {
-    "tags": []
-   },
+   "id": "af6a4710",
+   "metadata": {},
    "source": [
-    "### Exercise 2: Ice cream bowl 🌶️🌶️"
+    "### Exercise 3: Ice cream shop 🌶️🌶️🌶️"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "7f366029-6fa1-48f9-95ee-578b02addecd",
+   "id": "883794ff",
    "metadata": {},
    "source": [
-    "Create a class `Bowl` that can hold many ice cream scoops, as many as you like. You *should use* the custom class you created in the previous exercise.\n",
+    "Create a class `Shop` that sells many ice cream flavours. \n",
     "\n",
-    "The `Bowl` class should have a method called `add_scoops()` that accept **variable number** of scoops.\n",
+    "The `Shop` class should implement the comparison methods `__eq__`, `__lt__`, `__le__`.\n",
     "\n",
     "<div class=\"alert alert-block alert-info\">\n",
-    "    <h4><b>Hint</b></h4>\n",
-    "    In the <code>__init__</code> method of the <code>Bowl</code> class, you should define an attribute that acts as a container to hold the scoops you might want to add\n",
+    "    <h4><b>Hints</b></h4>\n",
+    "    <ul>\n",
+    "        <li>In the <code>__init__</code> method of the <code>Shop</code> class, you should define an attribute that acts as a container to hold the available flavours.</li>\n",
+    "        <li>You can use  <code>__eq__</code> and  <code>__lt__</code> to define <code>__le__</code>.</li>\n",
+    "    </ul>\n",
     "</div>\n",
     "\n",
     "<div class=\"alert alert-block alert-warning\">\n",
     "    <h4><b>Question</b></h4> \n",
-    "    Complete the solution function that creates a bowl of three scoops with flavors chocolate, vanilla, and stracciatella. The output of this function should be a <strong>string</strong> that reports the content of the bowl just created.\n",
-    "</div>\n",
-    "\n",
-    "For example:\n",
-    "\n",
-    "```\n",
-    "Ice cream bowl with chocolate, vanilla, stracciatella scoops\n",
-    "```"
+    "    Complete the solution function so that it creates two ice cream shops. <code>Shop 1</code> should sell the flavors provided by the parameter <code>flavor_1</code> and <code>Shop 2</code> should sell the flavors provided by the parameter <code>flavor_2</code>. The output of this function should be a <strong>boolean value</strong> which is the result of the following comparison: Is <code>Shop 1</code> selling a <strong>less or equal</strong> number of flavors to <code>Shop 2</code>?\n",
```

Comment:
```suggestion
    "    Complete the solution function so that it creates two ice cream shops. <code>Shop 1</code> should sell the flavors provided by the parameter <code>flavor_1</code> and <code>Shop 2</code> should sell the flavors provided by the parameter <code>flavor_2</code>. The output of this function should be a <strong>boolean value</strong>: Is <code>Shop 1</code> selling <strong>more or less</strong> flavors than <code>Shop 2</code>?\n",
```

Reply:
Ok, you're right: it changes the right answer to the exercise.

Maybe something like that?

```suggestion
    "    Complete the solution function so that it creates two ice cream shops. <code>Shop 1</code> should sell the flavors provided by the parameter <code>flavor_1</code> and <code>Shop 2</code> should sell the flavors provided by the parameter <code>flavor_2</code>. The output of this function should be a <strong>boolean value</strong> which is the result of the following comparison: Does <code>Shop 1</code> sell <strong>fewer or the same</strong> number of flavors as <code>Shop 2</code>?\n",
```

Reply:
Looks good, thanks a lot!